File: pascalsTriangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    if(numRows == 1) return [[1]];
    if(numRows == 2) return [[1],[1,1]];

    let answer = [[1],[1,1]];

    for(let answerIndex=1;answerIndex<numRows-1;answerIndex++){
    
        let pushArray = [], pushArrayIndex = 0;
        let i=0, j=1;
        while(pushArrayIndex<=answer[answerIndex].length){
            
            if(pushArrayIndex == 0 || pushArrayIndex == answer[answerIndex].length){
                pushArray.push(1);
                pushArrayIndex++;
            }else {
                let item = answer[answerIndex][i] + answer[answerIndex][j];
                console.log(item,i,j);
                i++;j++;
                pushArray.push(item);
                pushArrayIndex++
            }
        }
        answer.push(pushArray);
    }

    return answer;

};

generate(5);

File: Java/src/PascalsTriangle.java
import java.util.ArrayList;
import java.util.List;

public class PascalsTriangle {
    public static List<List<Integer>> generatePT(int numRows) {
        List<List<Integer>> answer = new ArrayList<>();
        answer.add(new ArrayList<>(List.of(1)));
        for(int answerIndex = 0;answerIndex<numRows-1;answerIndex++){
            List<Integer> pushArray = new ArrayList<>();
            int i=0, j=1;
            for(int pIndex = 0; pIndex <= answer.get(answerIndex).size(); pIndex++ ){
                if(pIndex == 0 || pIndex == answer.get(answerIndex).size()){
                    pushArray.add(1);
                }else{
                    pushArray.add((answer.get(answerIndex).get(i) + answer.get(answerIndex).get(j)));
                    i++;j++;
                }
            }
            answer.add(pushArray);
        }

        return answer;

    }

    public static void main(String[] args){
        List<List<Integer>> array = new ArrayList<>();

        array = generatePT(5);

        System.out.println(array);
    }
}
